apps/common/lib/lexical/source_file/store.ex

```diff
@@ -145,6 +146,48 @@ defmodule Lexical.SourceFile.Store do
 
     defp normalize_error(:error), do: {:error, :not_open}
     defp normalize_error(e), do: e
+
+    @read_types [:sources, :temp, :any]
+    @write_types [:sources, :temp]
+    defp ets_fetch(type, key) when type in @read_types do
+      case :ets.match(@table_name, {key, type_selector(type), :"$1"}) do
+        [[value]] -> {:ok, value}
+        _ -> :error
+      end
+    end
+
+    defp ets_put(type, key, value) when type in @write_types do
+      :ets.insert(@table_name, {key, type, value})
+      :ok
+    end
+
+    defp ets_has_key?(type, key) when type in @read_types do
+      match_spec = {key, type_selector(type), :"$1"}
+
+      case :ets.match(@table_name, match_spec) do
+        [] -> false
+        _ -> true
+      end
+    end
+
+    defp ets_pop(type, key) when type in @write_types do
+      with {:ok, value} <- ets_fetch(type, key),
+           :ok <- ets_delete(type, key) do
+        value
+      else
+        _ ->
+          nil
+      end
+    end
+
+    defp ets_delete(type, key) when type in @write_types do
+      match_spec = {key, type_selector(type), :_}
```

Comment:
I think the type_selector only makes sense to search for :sources, :temp or both, but I think we can skip using it when writing, WDYT?